### PR TITLE
Concatenate output files

### PR DESCRIPTION
**Pull request by sgsutcliffe**
Concatenate not just the summary.tsv but all the csv/tsv files.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/staramr/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/staramr _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by sgsutcliffe** [comment]
The issue we had with `CSVTK_CONCAT` nf-core module was that `iridanext-plugin` downstream was classifys all files with `meta.id` tags as being a `sample` but we had been passing a modified `meta.id` to `CSVTK_CONCAT` to properly name the output file. Removing the `meta.id` from the nf-core module fixed this. Due to modification of the module we moved it tp the local directory of `/modules`

**Review by apetkau — CHANGES_REQUESTED** [review]
This looks really great. Thanks so much Steven 😄 

I have a few comments and suggestions below.

File: modules/local/staramr/main.nf [hunk 1]
```diff
@@ -11,15 +11,15 @@ process STARAMR_SEARCH {
...
-    tuple val(meta), path("*_results/results.xlsx")          , emit: results_xlsx
-    tuple val(meta), path("*_results/${meta.id}_summary.tsv"), emit: summary_tsv
-    tuple val(meta), path("*_results/detailed_summary.tsv")  , emit: detailed_summary_tsv
-    tuple val(meta), path("*_results/resfinder.tsv")         , emit: resfinder_tsv
-    tuple val(meta), path("*_results/plasmidfinder.tsv")     , emit: plasmidfinder_tsv
-    tuple val(meta), path("*_results/mlst.tsv")              , emit: mlst_tsv
-    tuple val(meta), path("*_results/settings.txt")          , emit: settings_txt
-    tuple val(meta), path("*_results/pointfinder.tsv")       , emit: pointfinder_tsv, optional: true
-    path "versions.yml"                                      , emit: versions
+    tuple val(meta), path("*_results/results.xlsx")                     , emit: results_xlsx
```
[the remark this answers is not in the document]
Comment (sgsutcliffe) [reply modules/local/staramr/main.nf:14]: Yes, and an extra benefit of this is that I can clean up this code block below:
```
    # Change the names of output files to contain the genome name (allows for CSVTK module to concatenate files downstream)
    mv ${prefix}_results/summary.tsv ${prefix}_results/${prefix}_summary.tsv
    mv ${prefix}_results/detailed_summary.tsv ${prefix}_results/${prefix}_detailed_summary.tsv
    mv ${prefix}_results/resfinder.tsv ${prefix}_results/${prefix}_resfinder.tsv
    mv ${prefix}_results/plasmidfinder.tsv ${prefix}_results/${prefix}_plasmidfinder.tsv
    mv ${prefix}_results/mlst.tsv ${prefix}_results/${prefix}_mlst.tsv
    [[ -f ${prefix}_results/pointfinder.tsv ]] && mv ${prefix}_results/pointfinder.tsv ${prefix}_results/${prefix}_pointfinder.tsv || echo "No pointfinder.tsv"

```
to
```
for f in ${prefix}_results/* ; do mv "\$f" \$(echo \$f | sed 's;/;/${prefix}_;'); done
```

File: modules/local/csvtk/concat/main.nf [hunk 2]
```diff
@@ -13,7 +13,7 @@ process CSVTK_CONCAT {
     val out_format
 
     output:
-    tuple val(meta), path("${prefix}.${out_extension}"), emit: csv
+    tuple path("${prefix}.${out_extension}"), emit: csv
```
[the remark this answers is not in the document]
Comment (sgsutcliffe) [reply modules/local/csvtk/concat/main.nf:16]: That makes sense, but it is weird that my runs didn't give this error.

File: tests/main.nf.test [hunk 3]
```diff
@@ -26,9 +26,9 @@ nextflow_pipeline {
...
-            assert salmonella_metadata."CGE Predicted Phenotype" == "Spectinomycin, Streptomycin, Neomycin, Kanamycin, Lividomycin, Paromomycin, Ribostamycin, Amoxicillin, Amoxicillin+Clavulanic acid, Ampicillin, Ampicillin+Clavulanic acid, Cefotaxime, Cefoxitin, Ceftazidime, Piperacillin, Piperacillin+Tazobactam, Ticarcillin, Ticarcillin+Clavulanic acid, Cephalothin, Chloramphenicol, Trimethoprim, Benzylkonium Chloride, Ethidium Bromide, Chlorhexidine, Cetylpyridinium Chloride, Sulfamethoxazole, Doxycycline, Tetracycline"
-            assert salmonella_metadata."Genotype" == "aadA1, aadA2, aadA2, aph(3'')-Ib, aph(3')-Ia, blaCMY-2, blaTEM-1B, cmlA1, dfrA12, qacE, sul1, sul3, tet(A)"
+            assert salmonella_metadata."Predicted Phenotype" == "streptomycin, kanamycin, ampicillin, amoxicillin/clavulanic acid, cefoxitin, ceftriaxone, chloramphenicol, trimethoprim, ciprofloxacin I/R, nalidixic acid, unknown[qacE_1_X68232], sulfisoxazole, tetracycline"
```
[the remark this answers is not in the document]
Comment (sgsutcliffe) [reply tests/main.nf.test:29]: I think you're right the order wont always be the same. I'll start with the sorted version.

**Comment by sgsutcliffe** [comment]
The  `nf-test` files for checking `CSVTK_concat` are TSV files with empty columns in some rows (causing Trailing whitespace error) when running `pre-commit run` so I have added an exception for the these files to allow trailing whitespace.

File: workflows/staramr.nf [hunk 4]
```diff
@@ -67,17 +66,64 @@ workflow STARAMR {
...
     }
 
-    CSVTK_CONCAT (
-        ch_tsvs,
+    // 2) detailed_summary.tsv file
+    tsv_files_2 = STARAMR_SEARCH.out.detailed_summary_tsv
+
+    ch_tsvs_2 = tsv_files_2.map{
+        meta, detailed_summary_tsv -> detailed_summary_tsv
+    }.collect().map{
```
[the remark this answers is not in the document]
Comment (sgsutcliffe) [reply workflows/staramr.nf:85]: I think this is a great idea. I'd made a note to clean up this code, but with your suggestions I can easily finish it in this PR.

[the remark this answers is not in the document]
Comment (apetkau) [reply workflows/staramr.nf:85]: Thanks for posting as an issue: https://github.com/phac-nml/staramrnf/issues/16

**Review by apetkau — APPROVED** [review]
This looks great. Thanks so much for all your work on this Steven 😄

**Pull request merged**